sql-plugin/src/main/scala/com/nvidia/spark/rapids/literals.scala

```diff
@@ -734,6 +734,7 @@ case class GpuLiteral (value: Any, dataType: DataType) extends GpuLeafExpression
       case LongType => ast.Literal.ofLong(value.asInstanceOf[java.lang.Long])
       case FloatType => ast.Literal.ofFloat(value.asInstanceOf[java.lang.Float])
       case DoubleType => ast.Literal.ofDouble(value.asInstanceOf[java.lang.Double])
+      case StringType => ast.Literal.ofString(value.asInstanceOf[UTF8String].toString)
```

Comment:
I filed https://github.com/rapidsai/cudf/issues/14397 related to this. It might be nice to have a follow on issue where we don't bother to go from UTF8String -> Java String (in UCS2) -> UTF8-byte array.  We could just get the byte array from UTF8String and pass it directly to the Literal without any transcosing.

Reply:
Filed a PR addressing this in cudf side. https://github.com/rapidsai/cudf/pull/14402